src/vec_common_ppc.h

```diff
@@ -28,14 +28,173 @@
+ * scalar types that are not vector types:
+ * - __int128 (a general purpose register pair)
+ * - _Decimal128 (a floating-point even/odd register pair)
+ * - __ibm128 (a floating-point register pair)
+ * - __float128 (a vector register)
+ *
+ * These are not cast nor assignment compatible with any vector type.
+ * However it may be useful to transfer to/from vector types for
+ * conversion or manipulation within an operation.
+ * For example:
+ * - Conversions between __float128 and __int128, __ibm128,
+ * and _Decimal128 types.
+ * - Conversions between vector BCD integers and __int128
+ * and _Decimal128 types.
+ * - Conversions between vector __int128 and __float128, __ibm128,
+ * and _Decimal128 types.
+ *
+ * Here we use the __VEC_U_128 union to affect the transfer between
+ * the various types. We assume (fervently hope) that the compiler will
+ * recognize and optimize these as registers to registers transfers
+ * using the hardware instructions provided.
+ *
+ * The vector to/from __float128 transfer should be the simplest
+ * as __float128 operations are defined over the vector register set.
+ * However __float128 types are defined in the PowerISA and OpenPOWER
+ * ABI,  as scalars that just happens to use vector
+ * registers for parameter passing and operations.
+ * This distinction between scalars and vector prevents a direct cast
+ * between types. The __VEC_U_128 union is the simplest work around
+ * but in most cases no code should generated for this transfer.
+ * For example: vec_xfer_bin128_2_vui128t() and vec_xfer_vui128t_2_bin128().
+ *
+ * Any vector to/from __int128 transfer requires a transfer between
+ * vector and general purpose registers. POWER8 (PowerISA 2.07B) added
+ * Move to/from Vector Scalar Register (mfvsr, mtvsr) instructions.
+ * Again the __VEC_U_128 union is used to effect the transfer and the
+ * compiler should leverage the move instructions in the generate code.
+ *
+ * Any vector to/from __ibm128 or _Decimal128 requires a transfer
+ * between a pair of FPRs and a Vector Scalar Register (VSR).
+ * Technically this is transfer between the upper doubleword of two
+ * VSRs in the lower bank (VSR0-31) and another VSR.
+ * POWER7 (PowerISA 2.06B) provides the
+ * Permute Doubleword Immediate (xxpermdi) instruction.
+ * Again the __VEC_U_128 union is used to effect the transfer and the
+ * compiler should leverage the Permute Doubleword Immediate
+ * instructions in the generate code.
+ * For example: vec_BCD2DFP() and vec_DFP2BCD().
+ *
+ * \section common_endian_issues_0_0 Endian and vector constants
+ *
+ * Vector constants are often need to for; masking operations,
+ * range checks, permute selection, and radix conversion.
+ * Also compiler support for large integer and floating-point
+ * constants may be limited by the compiler.
+ * For example the GCC compilers support the (vector) __int128 type
+ * but do not directly support __int128 (39 digit) decimal constants.
```

Comment:
If this may change, you might specify something like "As of GCC 9, there is support for ... but not for ..."

Reply:
Add "This was verified in GCC 8.2," to end of the following \note